diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f456b9..c1f531b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -220,6 +220,7 @@ else()
     endif()
     set(BUILD_CURL_EXE OFF CACHE INTERNAL "" FORCE)
     set(BUILD_TESTING OFF)
+    set(ENABLE_WEBSOCKETS ON CACHE INTERNAL ON FORCE)
 
     if (CURL_VERBOSE_LOGGING)
         message(STATUS "Enabled curl debug features")
@@ -275,8 +276,8 @@ else()
         cmake_policy(SET CMP0135 NEW)
     endif()
     FetchContent_Declare(curl
-                         URL                    https://github.com/curl/curl/releases/download/curl-7_80_0/curl-7.80.0.tar.xz
-                         URL_HASH               SHA256=a132bd93188b938771135ac7c1f3ac1d3ce507c1fcbef8c471397639214ae2ab # the file hash for curl-7.80.0.tar.xz
+                         URL                    https://github.com/curl/curl/releases/download/curl-8_1_2/curl-8.1.2.tar.xz
+                         URL_HASH               SHA256=31b1118eb8bfd43cd95d9a3f146f814ff874f6ed3999b29d94f4d1e7dbac5ef6 # the file hash for curl-8.1.2.tar.xz
                          USES_TERMINAL_DOWNLOAD TRUE)   # <---- This is needed only for Ninja to show download progress
     FetchContent_MakeAvailable(curl)
 
